shims/spark300/src/main/scala/com/nvidia/spark/rapids/shims/spark300/GpuParquetScan.scala

```diff
@@ -42,10 +42,12 @@ case class GpuParquetScan(
     partitionFilters: Seq[Expression],
     dataFilters: Seq[Expression],
     rapidsConf: RapidsConf,
-    supportsSmallFileOpt: Boolean = true)
+    supportsMultiFileOpt: Boolean = true,
+    canUseMultiThreadRead: Boolean = true,
+    canUseCoalesceFilesRead: Boolean = true)
```

Comment:
this is here because of the InputFileName changes. If we fine hte user doing that, we copy the Exec and turn off canUseCoalesceFilesRead when we hit it.  I ended up only needing canUseCoalesceFilesRead for that, so I could remove the other 2 but left for consistency on how we pick them up. I'm fine with removing the other 2 though, thoughts?

Reply:
I'll rework it some to perhaps only have 1 parameter and make it more specific to queryUsesInputFileName